pkg/modules/pdfengines/routes.go

```diff
@@ -254,6 +254,25 @@ func WriteMetadataStub(ctx *api.Context, engine gotenberg.PdfEngine, metadata ma
 	return nil
 }
 
+// ProtectPdfWithPasswordStub adds password protection to PDF files.
+func ProtectPdfWithPasswordStub(ctx *api.Context, engine gotenberg.PdfEngine, userPassword, ownerPassword string, inputPaths []string) ([]string, error) {
+	if userPassword == "" {
+		return inputPaths, nil
+	}
+
+	outputPaths := make([]string, len(inputPaths))
+	for i, inputPath := range inputPaths {
+		outputPaths[i] = ctx.GeneratePath(".pdf")
+
+		err := engine.ProtectWithPassword(ctx, ctx.Log(), inputPath, outputPaths[i], userPassword, ownerPassword)
+		if err != nil {
+			return nil, fmt.Errorf("protect PDF '%s' with password: %w", inputPath, err)
+		}
+	}
+
+	return outputPaths, nil
+}
```

Comment:
We should keep the filename instead of generating a new one.

Is it possible to use the inputs in the engines instead? For instance, this is what has been done for flattening PDF:

```go
unc (engine *QPdf) Flatten(ctx context.Context, logger *zap.Logger, inputPath string) error {
	var args []string
	args = append(args, inputPath)
	args = append(args, "--generate-appearances")
	args = append(args, "--flatten-annotations=all")
	args = append(args, "--replace-input")
	args = append(args, engine.globalArgs...)
```